common/types.go

```diff
@@ -58,6 +61,9 @@ func (h *Hash) UnmarshalJSON(input []byte) error {
 	if length >= 2 && input[0] == '"' && input[length-1] == '"' {
 		input = input[1 : length-1]
 	}
+	if len(input) != 32 {
```

Comment:
Ok, this won't work :D The Input is hex encoded so it's 64 characters to start with. The it may or may not be previxed by `0x` so that must also be handled :P